junit-platform-engine/src/main/java/org/junit/platform/engine/reporting/FileEntry.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2015-2023 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.engine.reporting;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.util.Preconditions;
+import org.junit.platform.commons.util.ToStringBuilder;
+
+/**
+ * {@code FileEntry} encapsulates a file to be published to the reporting infrastructure.
+ *
+ * @since 1.11
+ * @see #from(Path)
+ */
+@API(status = EXPERIMENTAL, since = "1.11")
+public final class FileEntry {
```

Comment:
Debating whether we'd want some kind of MIME type, or if we'd trust report users to use extensions or content analysis to know which files were, for example, displayable as images...

Reply:
I can see a case for an optional mime type but making it mandatory is probably too heavy a burden on users. I'd expect one would browse the output dir in a file explorer or web browser at some point which will rely mostly on file extensions anyway. We could add a mime type later or right away. WDYT?

Reply:
Leaving it for later seems fine.